service/src/main/java/bio/terra/drshub/services/SignedUrlService.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.drshub.services;
+
+import bio.terra.bond.model.SaKeyObject;
+import bio.terra.common.iam.BearerToken;
+import bio.terra.drshub.config.DrsHubConfig;
+import bio.terra.drshub.logging.AuditLogEvent;
+import bio.terra.drshub.logging.AuditLogEventType;
+import bio.terra.drshub.logging.AuditLogger;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import java.net.URL;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public record SignedUrlService(
+    DrsHubConfig drsHubConfig,
+    AuthService authService,
+    DrsProviderService drsProviderService,
+    GoogleStorageService googleStorageService,
+    AuditLogger auditLogger) {
+
+  public URL getSignedUrl(
+      String bucket,
+      String objectName,
+      String dataObjectUri,
+      String googleProject,
+      BearerToken bearerToken,
+      String ip) {
+
+    var components = drsProviderService.getUriComponents(dataObjectUri);
+    var drsProvider = drsProviderService.determineDrsProvider(components);
+    SaKeyObject saKey = authService.fetchUserServiceAccount(drsProvider, bearerToken);
```

Comment:
If we are trying to mimic Martha's behavior, this needs to fall back to using the user's Pet SA if they are not authenticated with Bond.  https://github.com/broadinstitute/martha/blob/4e699ca160350b2702aef72201600e8ab91bd6c7/handlers/getSignedUrlV1.js#L12
https://github.com/broadinstitute/martha#get-signed-url

This may not be the ultimate behavior we want, but if we are just trying to port the Martha behavior to Bond, for now we should make it work this way.

Reply:
We specifically decided against replicating the Martha behavior exactly in [this ticket](https://broadworkbench.atlassian.net/browse/ID-169). Sam signs URLs for gs paths not backed by DRS.